src/main/java/io/github/cottonmc/resources/block/DefaultBlocks.java

```diff
@@ -1,22 +1,68 @@
 package io.github.cottonmc.resources.block;
 
+import java.util.Map;
+import java.util.function.Supplier;
+
+import com.google.common.collect.ImmutableMap;
+
 import io.github.cottonmc.cotton.registry.CommonBlocks;
-import net.minecraft.block.*;
-import net.minecraft.block.Block.*;
+import net.fabricmc.fabric.block.FabricBlockSettings;
+import net.fabricmc.fabric.tags.FabricItemTags;
+import net.minecraft.block.Block;
+import net.minecraft.block.Material;
+import net.minecraft.block.OreBlock;
 
 public class DefaultBlocks {
+    /** A generic metal ore supplier, similar to the properties of coal or iron ore. */
+    private static final Supplier<Block> STONE_TIER_ORE = ()->new OreBlock(FabricBlockSettings.of(Material.STONE)
+            .hardness(3.0f)
+            .resistance(3.0f)
+            .build());
+    /** A resilient metal ore provider, similar to the properties of redstone ore. */
+    private static final Supplier<Block> IRON_TIER_ORE = ()->new OreBlock(FabricBlockSettings.of(Material.STONE)
+            .hardness(3.0f)
+            .resistance(3.0f)
+            .breakByTool(FabricItemTags.PICKAXES, 2)
+            .build());
+    /** A generic metal block supplier, based on the properties from iron_block */
+    private static final Supplier<Block> METAL_BLOCK = ()->new Block(FabricBlockSettings.of(Material.METAL)
+            .hardness(5.0f)
+            .resistance(6.0f)
+            .build());
+
+    private static final Map<String, Supplier<Block>> BUILTINS = ImmutableMap.<String, Supplier<Block>>builder()
```

Comment:
Are we sure that we want an ImmutableMap? For all we know another mod might want to extend our default resources. Don't think it's likely, but something to keep in mind.
Like how the builder looks though. Far more elegant than the initial system.

Reply:
"extend our default resources"

you mean like making calls to Cotton instead of cotton-resources. Not altering this map.

Reply:
Nah, In case someone gets the idea to make a "cotton-resources+" mod that adds more possible default resources. Don't think it's likely, tbh, but just something I noticed.

Reply:
I think I thought of a possible use case: Let's say cotton-energy adds some wires. Now some people would need these wires while others would only want to use the cotton energy API. We could solve this problem either by splitting cotton-energy (api/implementation) or by adding the wires to cotton-resources' map of possible items. This way the wires will only available if a mod requests them to be enabled.